File: EX2/Conexion RESTful Transitsoft/ServidorRESTFul/TransitSoftBusiness/src/main/java/pe/edu/pucp/transitsoft/bo/server/VehiculoPropietarioBO.java
package pe.edu.pucp.transitsoft.bo.server;

import java.util.ArrayList;
import pe.edu.pucp.transitsoft.dao.VehiculoPropietarioDAO;
import pe.edu.pucp.transitsoft.daoImp.VehiculoPropietarioDAOImpl;
import pe.edu.pucp.transitsoft.model.VehiculoPropietarioDTO;
import pe.edu.pucp.transitsoft.model.VehiculoDTO;
import pe.edu.pucp.transitsoft.model.PropietarioDTO;

public class VehiculoPropietarioBO {

    private final VehiculoPropietarioDAO vehiculoPropietarioDAO;
    
    public VehiculoPropietarioBO(){
        this.vehiculoPropietarioDAO = new VehiculoPropietarioDAOImpl();
    }
    
    public Integer insertar(VehiculoPropietarioDTO vehiculoPropietarioDTO){
        return this.vehiculoPropietarioDAO.insertar(vehiculoPropietarioDTO);
    }
    
    public VehiculoPropietarioDTO obtenerPorId(Integer id){
        return this.vehiculoPropietarioDAO.obtenerPorId(id);
    }
    
    public ArrayList<VehiculoPropietarioDTO> listarTodos(){
        return this.vehiculoPropietarioDAO.listarTodos();
    }
    
    public Integer modificar(VehiculoPropietarioDTO vehiculoPropietarioDTO){
        return this.vehiculoPropietarioDAO.modificar(vehiculoPropietarioDTO);
    }
    
    public Integer eliminar(Integer id){
        VehiculoPropietarioDTO vehiculoPropietarioDTO = new VehiculoPropietarioDTO();
        vehiculoPropietarioDTO.setId(id);
        return this.vehiculoPropietarioDAO.eliminar(vehiculoPropietarioDTO);
    }
	
    public ArrayList<VehiculoDTO> listarVehiculosPorPropietario(Integer id_propietario){
        return this.vehiculoPropietarioDAO.listarVehiculosPorPropietario(id_propietario);
    }
    
    public ArrayList<PropietarioDTO> listarPropietariosPorVehiculo(Integer id_vehiculo){
        return this.vehiculoPropietarioDAO.listarPropietariosPorVehiculo(id_vehiculo);
    }
}

File: EX2/Conexion SOAP Transitsoft/TransitSoft/TransitSoftBusiness/VehiculoPropietarioBO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TransitSoftBusiness.VehiculoPropietarioWS; // <- Usa el namespace generado para tu WSDL de VehiculoPropietario

namespace TransitSoftBusiness
{
    public class VehiculoPropietarioBO
    {
        private VehiculoPropietarioClient clienteSOAP;

        public VehiculoPropietarioBO()
        {
            // Inicializa el cliente SOAP generado por Visual Studio
            this.clienteSOAP = new VehiculoPropietarioClient();
        }

        // ===================== CRUD =====================

        public int Insertar(int id_vehiculo, int id_propietario)
        {
            return this.clienteSOAP.insertarRelacion(id_vehiculo, id_propietario);
        }

        public int Modificar(int id, int id_vehiculo, int id_propietario)
        {
            return this.clienteSOAP.modificarRelacion(id, id_vehiculo, id_propietario);
        }

        public int Eliminar(int id)
        {
            return this.clienteSOAP.eliminarRelacion(id);
        }

        public vehiculoPropietarioDTO ObtenerPorId(int id)
        {
            return this.clienteSOAP.obtenerRelacionPorId(id);
        }

        public IList<vehiculoPropietarioDTO> ListarTodos()
        {
            return this.clienteSOAP.listarRelaciones();
        }

        // ===================== Consultas =====================

        public IList<vehiculoDTO> ListarVehiculosPorPropietario(int id_propietario)
        {
            return this.clienteSOAP.listarVehiculosPorPropietario(id_propietario);
        }

        public IList<propietarioDTO> ListarPropietariosPorVehiculo(int id_vehiculo)
        {
            return this.clienteSOAP.listarPropietariosPorVehiculo(id_vehiculo);
        }
    }
}